src/observable/ScalarObservable.ts

```diff
@@ -80,8 +80,8 @@ proto.reduce = function <T, R>(project: (acc: R, x: T) => R, seed?: R): Observab
     return <any>this;
   }
   let result = tryCatch(project)(seed, this.value);
-  if (result === errorObject) {
-    return new ErrorObservable(errorObject.e);
+  if (result as any === errorObject) {
```

Comment:
in some cases result is already any, in the cases where it isn't, we have to.

![image](https://cloud.githubusercontent.com/assets/1269157/11826060/ed8e739e-a350-11e5-8aed-a523dc976f01.png)

We could change `errorObject` to just be type any, and then this goes away.